interface/wx/memoryallocator.h

```diff
@@ -0,0 +1,98 @@
+/////////////////////////////////////////////////////////////////////////////
+// Name:        memoryallocator.h
+// Purpose:     interface of wxMemoryAllocator
+// Author:      wxWidgets team
+// Licence:     wxWindows licence
+/////////////////////////////////////////////////////////////////////////////
+
+/**
+    @class wxMemoryAllocator
+
+    This is a little helper class that hides the difference between
+    C++ operators new/delete and C way of memory management using
+    malloc()/free(). It is currently being used in wxImage to do
+    memory management. Right now it has 3 direct descendants - 1
+    for C-way, 1 for C++-way and 1 that will not throw on allocation.
+
+    @since 3.1.1
+*/
+class wxMemoryAllocator
+{
+public:
+    /**
+        Destructor.
+    */
+    virtual ~wxMemoryAllocator();
+    
+    /**
+        Default memory allocation function. This function may
+        either return NULL or throw std::bad_alloc on error.
+    */
+    void *Alloc(size_t size) = 0;
+   
+    /**
+        Default memory allocation function. This function should not throw.
+        It may be called with NULL pointer
+    */
+    void Free(void *ptr) = 0;
+};
+
+/**
+    This class represented memory management using C-way
```

Comment:
I believe that for all three classes, "represented memory management" is not correct. At least, the verb should be in simple present tense but perhaps the whole fragment could be worded better...

Reply:
@PBfordev ,
Are you native English speaking person? Can you worded this better?

Reply:
I am neither native English speaker nor any good at English.

I originally considered suggesting "This class manages memory in a C way" but it did not look good to me.

Reply:
@PBfordev ,
Then I'd consider wait for Vadim to chime in.
Hopefully he will have some idea of proper wording (even though he is not native English speaker either).
Or maybe someone else can word it better.

Reply:
How about "This class implements memory management in a C-style..."

Reply:
Changed as suggested.